tests/default_vars.sh

```diff
@@ -7,404 +7,406 @@
-  THRD_cpl_atmw_gdas=2; WPG_cpl_atmw_gdas=24; APB_cpl_atmw_gdas="0 311"; WPB_cpl_atmw_gdas="312 559"
-
-  TASKS_cpl_c96=192; TPN_cpl_c96=28; INPES_cpl_c96=3; JNPES_cpl_c96=8
-  THRD_cpl_c96=1; WPG_cpl_c96=6;  MPB_cpl_c96="0 143"; APB_cpl_c96="0 149"
-  CHM_cpl_c96="0 143"; OPB_cpl_c96="150 179"; IPB_cpl_c96="180 191"
-  NPROC_ICE_cpl_c96=12
-
-  TASKS_cpl_dflt=196; TPN_cpl_dflt=28; INPES_cpl_dflt=3; JNPES_cpl_dflt=8
-  THRD_cpl_dflt=1; WPG_cpl_dflt=6;  MPB_cpl_dflt="0 143"; APB_cpl_dflt="0 149"
-  CHM_cpl_dflt="0 143"; OPB_cpl_dflt="150 169"; IPB_cpl_dflt="170 177"; WPB_cpl_dflt="178 195"
-  NPROC_ICE_cpl_dflt=8
-
-  TASKS_cpl_thrd=120; TPN_cpl_thrd=14; INPES_cpl_thrd=3; JNPES_cpl_thrd=4
-  THRD_cpl_thrd=2; WPG_cpl_thrd=6;  MPB_cpl_thrd="0 71"; APB_cpl_thrd="0 77"
-  CHM_cpl_thrd="0 71"; OPB_cpl_thrd="78 97"; IPB_cpl_thrd="98 107"; WPB_cpl_thrd="108 119"
-  NPROC_ICE_cpl_thrd=10
-
-  TASKS_cpl_dcmp=196; TPN_cpl_dcmp=28; INPES_cpl_dcmp=4; JNPES_cpl_dcmp=6
-  THRD_cpl_dcmp=1; WPG_cpl_dcmp=6;  MPB_cpl_dcmp="0 143"; APB_cpl_dcmp="0 149"
-  CHM_cpl_dcmp="0 143"; OPB_cpl_dcmp="150 169"; IPB_cpl_dcmp="170 177"; WPB_cpl_dcmp="178 195"
-  NPROC_ICE_cpl_dcmp=8
+THRD=1
 
-  TASKS_cpl_mpi=280; TPN_cpl_mpi=28; INPES_cpl_mpi=4; JNPES_cpl_mpi=8
-  THRD_cpl_mpi=1; WPG_cpl_mpi=6;  MPB_cpl_mpi="0 191"; APB_cpl_mpi="0 197"
-  CHM_cpl_mpi="0 191"; OPB_cpl_mpi="198 231"; IPB_cpl_mpi="232 251"; WPB_cpl_mpi="252 279"
-  NPROC_ICE_cpl_mpi=20
-
-  TASKS_cpl_c384=480; TPN_cpl_c384=14; INPES_cpl_c384=6; JNPES_cpl_c384=8
-  THRD_cpl_c384=2; WPG_cpl_c384=24; MPB_cpl_c384="0 287"; APB_cpl_c384="0 311"
-  CHM_cpl_c384="0 287"; OPB_cpl_c384="312 431"; IPB_cpl_c384="432 479"
-  NPROC_ICE_cpl_c384=48
-
-  TASKS_cpl_bmrk=560; TPN_cpl_bmrk=28; INPES_cpl_bmrk=6; JNPES_cpl_bmrk=8
-  THRD_cpl_bmrk=1; WPG_cpl_bmrk=24; MPB_cpl_bmrk="0 287"; APB_cpl_bmrk="0 311"
-  CHM_cpl_bmrk="0 287"; OPB_cpl_bmrk="312 431"; IPB_cpl_bmrk="432 479"; WPB_cpl_bmrk="480 559"
-  NPROC_ICE_cpl_bmrk=48
+if [[ $MACHINE_ID = wcoss_cray ]]; then
 
-  TASKS_cpl_bmrk_aero=560; TPN_cpl_bmrk_aero=14; INPES_cpl_bmrk_aero=6; JNPES_cpl_bmrk_aero=8
-  THRD_cpl_bmrk_aero=2; WPG_cpl_bmrk_aero=24; MPB_cpl_bmrk_aero="0 287"; APB_cpl_bmrk_aero="0 311"
-  CHM_cpl_bmrk_aero="0 287"; OPB_cpl_bmrk_aero="312 431"; IPB_cpl_bmrk_aero="432 479"; WPB_cpl_bmrk_aero="480 559"
-  NPROC_ICE_cpl_bmrk_aero=48
+  TPN=24
 
-  TASKS_cpl_bmrk_mpi=600; TPN_cpl_bmrk_mpi=28; INPES_cpl_bmrk_mpi=6; JNPES_cpl_bmrk_mpi=8
-  THRD_cpl_bmrk_mpi=1; WPG_cpl_bmrk_mpi=24; MPB_cpl_bmrk_mpi="0 287"; APB_cpl_bmrk_mpi="0 311"
-  OPB_cpl_bmrk_mpi="312 431"; IPB_cpl_bmrk_mpi="432 479"; WPB_cpl_bmrk_mpi="480 599"
-  NPROC_ICE_cpl_bmrk_mpi=48
-
-  TASKS_cpl_c192=384; TPN_cpl_c192=28; INPES_cpl_c192=6; JNPES_cpl_c192=8
-  THRD_cpl_c192=1; WPG_cpl_c192=12;  MPB_cpl_c192="0 287"; APB_cpl_c192="0 299"
-  CHM_cpl_c192="0 287"; OPB_cpl_c192="300 359"; IPB_cpl_c192="360 383"
-  NPROC_ICE_cpl_c192=24
+  TASKS_dflt=150 ; INPES_dflt=3 ; JNPES_dflt=8
```

Comment:
Ideally, TASKS does not need to be specified since TASKS is not independent of (INPES, JNPES, NTILES, WRITE_GROUP, WRTTASK_PER_GROUP).

Reply:
I actually tried to remove TASKS but the change broke rocoto. Unless we restructure the computation of total tasks somehow. For example by moving function that computes tasking information outside run_test.sh. I decided to not make that change now until we decide what we are going to do about 3 platforms I have no access to run tests on (s4, stampede, expanse). Maybe in one of the future PRs. I left a comment in run_test.sh where I compare $TASKS to $UFS_tasks.